e2e/sharedFlows/resetUserForTesting.js

```diff
@@ -0,0 +1,159 @@
+
+const apiHost = Config.OAUTH_API_URL;
+
+const testUsernameAllowlist = [
+  "inaturalist-test"
+];
+
+const userAgent = "iNaturalistRN/e2e";
+const installId = uuid.v4( ).toString( );
+
+inatjs.setConfig( {
+  apiURL: Config.API_URL,
+  writeApiURL: Config.API_URL,
+  userAgent,
+  headers: {
+    "X-Installation-ID": installId
+  }
+} );
+
+// programatically dismisses announcements for user and resets to a lone sample observation
+// in order to set up consistent testing conditions and remove need to wait for announcements
+export default async function resetUserForTesting() {
+  console.log(
+    "Test user reset: dismissing announcements and resetting observations..."
+  );
+
+  if ( !testUsernameAllowlist.includes( Config.E2E_TEST_USERNAME ) ) {
+    const message = "This e2e test deletes observations of the user under test."
+    + "Add this username to the `testUsernameAllowlist` if that's really what you want.";
+    throw new Error( message );
+  }
+
+  const apiClient = create( {
+    baseURL: apiHost,
+    headers: {
+      "User-Agent": userAgent,
+      "X-Installation-ID": installId
+    }
+  } );
+
+  await apiClient.get( "/logout" );
+
+  const formData = {
+    format: "json",
+    grant_type: "password",
+    client_id: Config.OAUTH_CLIENT_ID,
+    client_secret: Config.OAUTH_CLIENT_SECRET,
+    username: Config.E2E_TEST_USERNAME,
+    password: Config.E2E_TEST_PASSWORD,
+    locale: "en"
+  };
+
+  const tokenResponse = await apiClient.post( "/oauth/token", formData );
+  const accessToken = tokenResponse.data.access_token;
+
+  apiClient.setHeader( "Authorization", `Bearer ${accessToken}` );
+
+  const jwtResponse = await apiClient.get( "/users/api_token.json" );
+
+  const opts = {
+    api_token: jwtResponse.data.api_token
+  };
+
+  const announcementSearchParams = {
+    placement: "mobile",
+    locale: "en-US",
+    per_page: 20,
+    fields: {
+      id: true,
+      dismissible: true
+    }
+  };
+
+  const announcementResponse = await inatjs.announcements.search(
+    announcementSearchParams,
+    opts
+  );
+
+  const announcementIdsToDismiss = announcementResponse
+    .results
+    .filter( a => a.dismissible )
+    .map( a => a.id );
+
+  console.log( `Dismissing ${announcementIdsToDismiss.length} announcements` );
+
+  await Promise.all( announcementIdsToDismiss.map( async id => {
+    await inatjs.announcements.dismiss(
+      { id },
+      opts
+    );
+  } ) );
+
+  const usersEditResponse = await apiClient.get(
+    "/users/edit.json",
+    {},
+    {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        "User-Agent": userAgent
+      }
+    }
+  );
+
+  const userId = usersEditResponse.data.id;
+
+  const observationResponse = await inatjs.observations.search( { user_id: userId }, opts );
+
+  // is this _really_ an e2e test user if they have more than expected observations?
+  const suspiciousObservationThreshold = 10;
+  if ( typeof observationResponse.total_results !== "number"
```

Comment:
I would make a different error message for `typeof observationResponse.total_results !== "number" `. I can imagine someone seeing the error message and without looking at the code trying to solve this by going to the website and not seeing any observations there.